### DISCUSSION TITLE
How to auto confirm `TSUpdate` ?

### DISCUSSION THREAD

**Discussion by yavorski** [nvim-treesitter/nvim-treesitter#4890] [Q&A]
Hi, is there an unobtrusive way to do `TSUpdate` or `TSUpdateSync` and escape/autoconfirm/yes user confirmation prompt? 
Can we auto pass confirm option somehow with `ensure_installed = "all"`? 
Using Lazy, here is example config and screenshot ... I have tried both `TSUpdate` and `TSUpdateSync` with no luck.

```lua
{
  "nvim-treesitter/nvim-treesitter",
  build = ":TSUpdate", -- or TSUpdateSync
  event = { "BufReadPost", "BufNewFile" },
  opts = {
    ensure_installed = "all",
    highlight = { enable = true },
    indent = { enable = true },
  },
  config = function(plugin, opts)
    require("nvim-treesitter.configs").setup(opts)
  end
}
```
![nvim-tree-sitter-all](https://github.com/nvim-treesitter/nvim-treesitter/assets/3436517/8d7c50e9-f6eb-491c-a024-03fe3c0ae689)

**Comment by clason** [comment]
There shouldn't be a prompt for `TSUpdate`. If you get constant reinstalling, there's an issue with how you load nvim-treesitter (**which does not support lazy-loading, especially on events**!)

**Comment by yavorski** [comment]
Hi, I don't get constant reinstalling. This is just the screen for the first time install e.g. on new machine setup. 
These messages triggers a pager ( --More -- ), in which you need to click <kbd>space</kbd> or <kbd>enter</kbd> in order to proceed with the installation, which is what I am trying to avoid. 
Hope that this makes sense.

**Reply by clason** [reply]
Not much you can do about that, I'm afraid -- except not installing all 190 parsers by default (which you are strongly advised not to do).

Have you tried the `auto_install` function?

**Reply by yavorski** [reply]
I tried with just 20 parsers only and it is the same -> "Press Enter or type command to continue". The `auto_install` makes no difference.
Anyway thanks for the help, it seems it is not possible without a dedicated option to opt out from the messages.

![nvim-tree-sitter-prompt-21](https://github.com/nvim-treesitter/nvim-treesitter/assets/3436517/75a0eefd-ddb1-40eb-bd74-a513f48cfb60)
